### PR TITLE
Requests file

### PR DESCRIPTION
**Pull request by anikamukherji**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -2,26 +2,60 @@
...
-    user.save() # save the user to the database
+    user = models.User.objects.raw({"_id": email}).first()
+    user.heart_rate.append(heart_rate)
+    user.heart_rate_times.append(time)
+    user.save()
+
+
+def create_user(email, age=None, hr=None):
+    u = models.User(email, age, [], [])
+    if hr:
```
Comment (suyashkumar) [inline main.py]: if `hr` is 0 this will be false-y and will not trigger. probably best to use `if hr is not None`

File: main.py [hunk 2]
```diff
@@ -2,26 +2,60 @@
...
+        user = models.User.objects.raw({"_id": email}).first()
+    else:
+        return None
+    return user.heart_rate
+
+
+def already_user(email):
+    try:
+        user = models.User.objects.raw({"_id": email}).first()
+    except:
```
Comment (suyashkumar) [inline main.py]: try to catch a particular Exception here.

File: main.py [hunk 3]
```diff
@@ -2,26 +2,60 @@
...
+    vals = {
+            "user_email": user.email,
+            "user_age": user.age,
+            "heart_rates": user.heart_rate,
+            "heart_rate_times": user.heart_rate_times
+            }
+    return vals
+
+
+def user_heart_rates(email):
```
Comment (suyashkumar) [inline main.py]: consider more descriptive "verb" like naming like "get_heart_rates"

File: main.py [hunk 4]
```diff
@@ -2,26 +2,60 @@
...
 
 def print_user(email):
-    user = models.User.objects.raw({"_id": "suyash@suyashkumar.com"}).first() # Get the first user where _id=suyash@suyashkumar.com
+    user = models.User.objects.raw({"_id": email}).first()
     print(user.email)
     print(user.heart_rate)
     print(user.heart_rate_times)
 
+
+def user_dict(email):
```
Comment (suyashkumar) [inline main.py:28]: you can define functions like this right on the `User` class actually :)

File: requests.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    Posts new user with given heart rate
+    """
+    r = request.get_json()
+    try:
+        email = r["user_email"]
+        age = r["user_age"]
+        hr = r["heart_rate"]
+    except KeyError as e:
+        logging.warning("Incorrect JSON input: {}".format(e))
+        return
```
Comment (suyashkumar) [inline requests.py]: probably want to return a 4xx error code and some sort of error response body

File: requests.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    """
+    Gets heart rates for given user email
+
+    :return: all heart rates for user
+    :rtype: dict
+    """
+    heart_rates = user_heart_rates(user_email)
+    if heart_rates is None:
+        logging.warning("User with email does not exist".format(user_email))
+        return
```
Comment (suyashkumar) [inline requests.py]: same comment as above with error codes

**Review by suyashkumar — COMMENTED** [review]
Left some comments, but overall good job, definitely on the right track!

File: main.py (hunk 2 above)
Comment (anikamukherji) [reply main.py]: Do you know what Exception is thrown if there is no user that exists with that id? I am having trouble finding source code online. Or, is there a `.exists()` or `.count()` function that can accompany a filter? I'm just having trouble finding documentation in general

Comment (anikamukherji) [reply main.py]: Just kidding I found a way with...
`return models.User.objects.raw({"_id": email}).count() > 0`

**Pull request merged**